include/unifex/await_transform.hpp

```diff
@@ -88,60 +120,60 @@ class _as_awaitable {
 
     template <typename Func>
     friend void
-    tag_invoke(tag_t<visit_continuations>, const receiver& r, Func&& func) {
+    tag_invoke(tag_t<visit_continuations>, const _rec& r, Func&& func) {
       visit_continuations(r.continuation_.promise(), (Func&&)func);
     }
 
   private:
-    _as_awaitable* op_;
+    _expected<Value>* result_;
     coro::coroutine_handle<Promise> continuation_;
   };
 
-public:
-  explicit _as_awaitable(Sender&& sender, coro::coroutine_handle<Promise> h)
-    noexcept(is_nothrow_connectable_v<Sender, receiver>)
-  : op_(unifex::connect((Sender&&)sender, receiver{this, h}))
-  {}
-
-  ~_as_awaitable() {
-    switch(state_) {
-    case state::value:
-      unifex::deactivate_union_member(value_);
-      break;
-    case state::exception:
-      unifex::deactivate_union_member(exception_);
-      break;
-    default:;
-    }
-  }
+  _expected<Value> result_;
 
+public:
   bool await_ready() const noexcept {
     return false;
   }
 
-  void await_suspend(coro::coroutine_handle<Promise>) noexcept {
-    unifex::start(op_);
-  }
-
-  value_t await_resume() {
-    switch (state_) {
+  Value await_resume() {
+    switch (result_.state_) {
     case state::value:
-      return std::move(value_).get();
+      return std::move(result_.value_).get();
     default:
-      assert(state_ == state::exception);
-      std::rethrow_exception(exception_.get());
+      assert(result_.state_ == state::exception);
+      std::rethrow_exception(result_.exception_.get());
     }
   }
+};
+
+template <typename Sender>
+using _single_sender_value_type_t =
```

Comment:
This would be better named `_sender_single_value_type_t`.
The current naming reads to me like it's the value_type of a single-sender (as opposed to a many-sender).

Also, maybe move this out to `sender_concepts.hpp` along with `sender_value_types_t`, `sender_error_types_t` type-aliases?